dcc-portal-api/src/main/java/org/icgc/dcc/portal/resource/core/DownloadResource.java

```diff
@@ -202,7 +169,16 @@ public JobInfo submitJob(
       ) {
     // Work out the query for that returns only donor ids that matches the filter conditions
     val donorIds = donorService.findIds(Query.builder().filters(filters.get()).build());
-    val dataTypeSizes = DownloadResources.normalizeSizes(downloadClient.getSizes(donorIds));
+    Map<DownloadDataType, Long> sizes = null;
+
+    try {
+      sizes = downloadClient.getSizes(donorIds);
+    } catch (Exception e) {
+      log.error("Exception while resolving download files size: \n", e);
+    }
+    checkRequest(sizes == null, "Failed to resolve files size.");
```

Comment:
What if there is a server side error? Maybe you need to be more explicit with the exception types.